### PR TITLE
[s] Fixed dragging shackled prison shoes off

### LINKED ISSUE #36318: Prison Shoes can be removed by yourself
[Round ID]: # 84750

[Testmerges]: # None

[Reproduction]: # Create prison shoes by attaching handcuffs to shoes. Put them on. Click with hand to remove them to get the standard "you need help removing this". Drag the item from feet to hand to remove it anyways.

### PR DESCRIPTION
**Pull request by Davidj361**
[Changelogs]: # (Your PR should contain a detailed changelog of notable changes, titled and categorized appropriately. This includes, new features, sprites, sounds, balance changes, admin tools, map edits, removals, big refactors, config changes, hosting changes and important fixes. An example changelog has been provided below for you to edit. If you need additional help, read https://github.com/tgstation/tgstation/wiki/Changelogs)

:cl:
fix: Fixed cuffed prison shoes being taken off by dragging them into your hand.
/:cl:

[why]:
Fixes #36318

### CODE REVIEW

**Comment by ninjanomnom** [comment]
http://thecodelesscode.com/case/123

**Comment by ShizCalev** [comment]
conflict.

**Comment by ExcessiveUseOfCobblestone** [comment]
@Davidj361 for future reference if you are going to fix exploits please add [s] to the title of your pr.

**Comment by Davidj361** [comment]
@ExcessiveUseOfCobblestone
>for future reference if you are going to fix exploits please add [s] to the title of your pr.

Why?

**Comment by ShizCalev** [comment]
[s] turns off the ingame alert for the pull request when it's initially opened, which is typically useful (and should only be used) for ingame exploits or server critical issues.

File: code/modules/clothing/shoes/colour.dm [hunk 1]
```diff
@@ -109,5 +109,16 @@
...
 		if(C.shoes == src && chained == 1)
 			to_chat(user, "<span class='warning'>You need help taking these off!</span>")
+			return FALSE
+	return ..()
+
+/obj/item/clothing/shoes/sneakers/orange/MouseDrop(atom/over)
+	var/mob/m = usr
+	if(ishuman(m))
+		var/mob/living/carbon/human/c = m
+		if(c.shoes == src && src.chained == 1)
```
Comment (ShizCalev) [inline code/modules/clothing/shoes/colour.dm]: remove `src.`

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by Davidj361** [comment]
Done

File: code/modules/clothing/shoes/colour.dm [hunk 2]
```diff
@@ -109,5 +109,16 @@
 		var/mob/living/carbon/human/C = user
 		if(C.shoes == src && chained == 1)
 			to_chat(user, "<span class='warning'>You need help taking these off!</span>")
+			return FALSE
+	return ..()
+
+/obj/item/clothing/shoes/sneakers/orange/MouseDrop(atom/over)
+	var/mob/m = usr
+	if(ishuman(m))
```
Comment (ninjanomnom) [inline code/modules/clothing/shoes/colour.dm]: You'll only be doing something here if `chained` is true anyway so check it before this type check.
It's only ever 1 or 0 aka TRUE or FALSE so do
`if(chained && ishuman(m))`

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Pull request merged**